artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/LargeServerMessageImpl.java

```diff
@@ -511,6 +512,44 @@ protected void closeFile() throws Exception {
       }
    }
 
+   private static int read(final SequentialFile file, final ByteBuffer bufferRead) throws Exception {
```

Comment:
Given this is a hot path and in most cases(LargeMessageDeliverer::deliver, ClientProducerImpl::largeMessageSendServer, CoreMessage::getLargeMessageBuffer, etc) "bufferRead" is a heap buffer, can we construct "bufferRead" by using PooledByteBufAllocator before calling to save from coping buffer?

Reply:
@wy96f 
It was the second point on my list :) 
Yes, it can be done, but deserve lot of attention and probably some changes on the API on SessionCallback...

Reply:
@wy96f I still think that `Page` has the same issue and should use Netty pools instead of retaining direct memory as a state

Reply:
Do you mean readFileBuffer in Page? In most cases, we sequentially read messages from page and readFileBuffer might contains several messages(depends on message size) in buffer. This way we can just read following messages from buffer(no need bothering file system). If we use netty pools, we need to read the file every time we read a message, right? Will this somewhat affect the perf?

Reply:
> If we use netty pools, we need to read the file every time we read a message, right? Will this somewhat affect the perf?
We need to find a proper solution, but is a leak and we use a heap buffer too there so it means a leak on IOUtil too...
I see the same on decoding side for replicated files and on encoding side for non netty connections...basically everytime we performa FileChannel ops with heap ByteBuffers.

Reply:
Agreed :)